server/src/main/java/io/whitefox/api/server/StorageV1ApiImpl.java

```diff
@@ -0,0 +1,62 @@
+package io.whitefox.api.server;
+
+import io.whitefox.api.deltasharing.Mappers;
+import io.whitefox.api.model.v1.generated.UpdateStorage;
+import io.whitefox.api.server.v1.generated.StorageV1Api;
+import io.whitefox.core.Principal;
+import io.whitefox.core.services.StorageService;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.Response;
+
+public class StorageV1ApiImpl implements StorageV1Api, ApiUtils {
+
+  private final StorageService storageService;
+
+  @Inject
+  public StorageV1ApiImpl(StorageService storageService) {
+    this.storageService = storageService;
+  }
+
+  @Override
+  public Response createStorage(io.whitefox.api.model.v1.generated.CreateStorage createStorage) {
+    return wrapExceptions(
+        () -> Response.status(Response.Status.CREATED)
+            .entity(Mappers.storage2api(storageService.createStorage(
+                Mappers.api2createStorage(createStorage, getRequestPrincipal()))))
+            .build(),
+        exceptionToResponse);
+  }
+
+  private Principal getRequestPrincipal() {
```

Comment:
we should factor this "service" somewhere

Reply:
yeah, I thought of putting it under ApiUtils for now since it is a mock impl